controllers/cloud.redhat.com/clowdenvironment_reconciliation.go

```diff
@@ -0,0 +1,489 @@
+	namespace := core.Namespace{}
+	namespaceName := types.NamespacedName{
+		Name: r.env.Spec.TargetNamespace,
+	}
+	if nErr := r.client.Get(*r.ctx, namespaceName, &namespace); nErr != nil {
+		r.log.Info("Namespace get error", "err", nErr)
+		r.recorder.Eventf(r.env, "Warning", "NamespaceMissing", "Requested Target Namespace [%s] is missing", r.env.Spec.TargetNamespace)
+		if setClowdStatusErr := SetClowdEnvConditions(*r.ctx, r.client, r.env, crd.ReconciliationFailed, nErr); setClowdStatusErr != nil {
+			r.log.Info("Set status error", "err", setClowdStatusErr)
+			return ctrl.Result{Requeue: true}, setClowdStatusErr
+		}
+		return ctrl.Result{Requeue: true}, nErr
+	}
+	r.env.Status.TargetNamespace = r.env.Spec.TargetNamespace
+	return ctrl.Result{}, nil
+}
+
+//Make a new target namespace
+func (r *ClowdEnvironmentReconciliation) _makeTargetNamespace() (ctrl.Result, error) {
+	r.env.Status.TargetNamespace = r.env.GenerateTargetNamespace()
+	namespace := &core.Namespace{}
+	namespace.SetName(r.env.Status.TargetNamespace)
+	if snErr := r.client.Create(*r.ctx, namespace); snErr != nil {
+		r.log.Info("Namespace create error", "err", snErr)
+		if setClowdStatusErr := SetClowdEnvConditions(*r.ctx, r.client, r.env, crd.ReconciliationFailed, snErr); setClowdStatusErr != nil {
+			r.log.Info("Set status error", "err", setClowdStatusErr)
+			return ctrl.Result{Requeue: true}, setClowdStatusErr
+		}
+		return ctrl.Result{Requeue: true}, snErr
+	}
+	return ctrl.Result{}, nil
+}
+
+//Update the target namespace
+func (r *ClowdEnvironmentReconciliation) _updateTargetNamespace() (ctrl.Result, error) {
+	if statErr := r.client.Status().Update(*r.ctx, r.env); statErr != nil {
+		r.log.Info("Namespace create error", "err", statErr)
+		if setClowdStatusErr := SetClowdEnvConditions(*r.ctx, r.client, r.env, crd.ReconciliationFailed, statErr); setClowdStatusErr != nil {
+			r.log.Info("Set status error", "err", setClowdStatusErr)
+			return ctrl.Result{Requeue: true}, setClowdStatusErr
+		}
+		return ctrl.Result{Requeue: true}, statErr
+	}
+	return ctrl.Result{}, nil
+}
+
+//Determine if the target namespace is marked for deletion
+//If it is then we enter into the SKIPRECONCILE corner case
+//which is a special case where the reconciliation controller bails early
+//from reconcile WIHTOUT returning an error
+func (r *ClowdEnvironmentReconciliation) isTargetNamespaceMarkedForDeletion() (ctrl.Result, error) {
+	ens := &core.Namespace{}
+	if getNSErr := r.client.Get(*r.ctx, types.NamespacedName{Name: r.env.Status.TargetNamespace}, ens); getNSErr != nil {
+		return ctrl.Result{Requeue: true}, getNSErr
+	}
+
+	if ens.ObjectMeta.DeletionTimestamp != nil {
+		r.log.Info("Env target namespace is to be deleted - skipping reconcile")
+		return ctrl.Result{}, errors.New(SKIPRECONCILE)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *ClowdEnvironmentReconciliation) runProviders() (ctrl.Result, error) {
+	provider := r._makeProvider()
```

Comment:
Is this just to reduce line count of the function? I'd perhaps argue in a case like this, that being able to see the literal would help understanding.

Reply:
No it was just being DRY. We create the provider in two places and if I notice something being done in 2 places I'll factor it out. None of the logic in the code deals with the internal state of the provider, we just pass it out somewhere so I figured this made sense.

Reply:
Discussed. Removed function